drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/cache/util/maps/ArrayMultiMap.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.guided.dtable.client.widget.analysis.cache.util.maps;
+
+import java.util.ArrayList;
+import java.util.SortedMap;
+
+class ArrayMultiMap<Key extends Comparable, Value>
```

Comment:
Hehe. The more I dig into the Guava source code the more I similarities I find. Perfect example how simple thing becomes complicated and then you were better going with something someone else did anyway.
Well, at least now it is clear what I need to override and extend and why.